File: src/main/java/com/github/cao/awa/majalis/env/human/organ/heart/heartbeat/wave/WaveMetadata.java
package com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave;

import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.shape.WaveRenderShape;
import lombok.Getter;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;

@Getter
@Setter
public abstract class WaveMetadata {
    @NotNull
    private final WaveRenderShape renderShape;
    private int peakedTime = 0;
    private int troughTime = 0;
    private int endTime = 0;

    public WaveMetadata(@NotNull WaveRenderShape renderShape) {
        this.renderShape = renderShape;
    }

    @NotNull
    public abstract WaveType type();
}

File: src/main/java/com/github/cao/awa/majalis/env/human/organ/heart/heartbeat/wave/WaveMetadata.kt
package com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave

import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.shape.WaveRenderShape

abstract class WaveMetadata(val renderShape: WaveRenderShape) {
    var peakedTime = 0
    var troughTime = 0
    var endTime = 0

    abstract fun type(): WaveType
}
